woocommerce/wc-admin
# Chart Component: Add/chart intervals

Chart Component Master Thread - https://github.com/woocommerce/wc-admin/issues/164

This PR addresses issue specifically raised in #320 , including custom `xFormat`, `yFormat` and `tooltipFormat` props.

![screenshot 2018-09-05 22 50 29](https://user-images.githubusercontent.com/1160732/45120634-0218f680-b15f-11e8-80d5-b0bea4488d1f.png)

### Testing instructions:
- update `dummy-hour.js` to monthly (or other) data, for example:
```
export default [
	{
		date: '2018-08-01T00:00:00',
		'Product X': 58929.99,
		'Product Y': 160130.74000000002,
	},
	{
		date: '2018-09-01T00:00:00',
		'Product X': 3805.56,
		'Product Y': 123876,
	},
	{
		date: '2018-10-01T00:00:00',
		'Product X': 3805.56,
		'Product Y': 218921,
	},
];
```

- update `example-hour.js` inline with the data above using the [d3 time formatting specifier](https://github.com/d3/d3-time-format#locale_format) of your choice for the x-axis and tooltips, for example:
```
import dummyOrders from './test/fixtures/dummy-hour';

...
<Chart
    data={ dummyOrders }
    tooltipFormat={ "%b '%y" }
    type={ 'bar' }
    xFormat={ '%b %y' }
/>
```

Note that you can include words and characters in the specifiers, for example, `tooltip={ 'ISO Week %V'}`.

- update the chart reference in `dasboard/index.js` to `import ChartExample from 'components/chart/example-hour';`
- make sure you see what you expect in the x-axis and tooltip 
- change to a line chart and make sure it works as expected
